environment

```diff
@@ -30,6 +30,13 @@ _select () {
 # https://github.com/boto/boto3/issues/1179#issuecomment-569940530
 #
 _preauth () {
+    if ! gcloud auth application-default print-access-token >/dev/null 2>&1; then
```

Comment:
I think this should be a loop so that the post condition is that a successful call `print-access-token` was made.